src/shogun/converter/ica/FastICA.cpp

```diff
@@ -128,9 +128,13 @@ CFeatures* CFastICA::apply(CFeatures* features)
 
 	W = sym_decorrelation(W);
 
-	int iter = 0;
 	float64_t lim = tol+1;
-	while (lim > tol && iter < max_iter)
+	for (auto i : progress(
+	    range(0, max_iter),
+	    "PROGRESS: ",
```

Comment:
@geektoni this seems a bit verbose. Isnt there a shorter version?

Reply:
Sorry! I just only realise this is the advanced way of using it

Reply:
@geektoni Could we have a way to pass lambda functions that does not require to pass the string and the encoding?
I.e. something as compact as the `for (auto i: progress(range(0, 10)))` but with a custom stopping condition?

Reply:
#4240

Reply:
@sunalbert could you rebase and use the new function?

Reply:
Ok, Thanks @iglesias @karlnapf

Reply:
Of course, you are welcome!